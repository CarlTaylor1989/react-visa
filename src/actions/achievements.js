import * as types from './achievementTypes';

export function setCompletedAchievement(achievement) {
  return {
    type: types.SET_COMPLETED_ACHIEVEMENT,
    achievement
  };
}

export function restoreAchievements(completed, lastCompleted) {
  return {
    type: types.RESTORE_ACHIEVEMENTS,
    completed,
    lastCompleted
  };
}

export function updateProductAchievements() {
  return {
    type: types.UPDATE_PRODUCT_ACHIEVEMENTS
  };
}

export function updateProposalAchievements() {
  return {
    type: types.UPDATE_PROPOSAL_ACHIEVEMENTS
  };
}

export function updateScoredAchievements() {
  return {
    type: types.UPDATE_SCORED_ACHIEVEMENTS
  };
}

export function updateSettingsAchievements() {
  return {
    type: types.UPDATE_SETTINGS_ACHIEVEMENTS
  };
}

export function updateTimedAchievements() {
  return {
    type: types.UPDATE_TIMED_ACHIEVEMENTS
  };
}

export function resetChangedAchievement() {
  return {
    type: types.RESET_CHANGED_ACHIEVEMENT
  };
}

export function resetAchievements() {
  return {
    type: types.RESET_ACHIEVEMENTS
  };
}
