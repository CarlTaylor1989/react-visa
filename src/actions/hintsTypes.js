export const SET_CURRENT_HINT_GROUP = 'SET_CURRENT_HINT_GROUP';
export const SET_HINT_GROUP_INDEX = 'SET_HINT_GROUP_INDEX';
export const SET_HINTS_PAUSED = 'SET_HINTS_PAUSED';
