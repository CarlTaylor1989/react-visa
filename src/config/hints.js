export const HINT_DIAGNOSTIC = 'diagnostic';
export const HINT_NETWORK = 'network';
export const HINT_PRODUCT = 'product';
export const HINT_CLIENT_SELECTION = 'clientselection';
export const HINT_CLIENT_PROPOSAL = 'clientproposal';

export const HINT_TIME = 20000;

export default [
  {
    id: 'h1',
    group: HINT_DIAGNOSTIC
  },
  {
    id: 'h2',
    group: HINT_NETWORK
  },
  {
    id: 'h3',
    group: HINT_NETWORK
  },
  {
    id: 'h4',
    group: HINT_NETWORK
  },
  {
    id: 'h5',
    group: HINT_NETWORK
  },
  {
    id: 'h6',
    group: HINT_NETWORK
  },
  {
    id: 'h7',
    group: HINT_NETWORK
  },
  {
    id: 'h8',
    group: HINT_NETWORK
  },
  {
    id: 'h9',
    group: HINT_PRODUCT
  },
  {
    id: 'h10',
    group: HINT_PRODUCT
  },
  {
    id: 'h11',
    group: HINT_CLIENT_SELECTION
  },
  {
    id: 'h12',
    group: HINT_CLIENT_PROPOSAL
  },
  {
    id: 'h13',
    group: HINT_CLIENT_PROPOSAL
  }
];
