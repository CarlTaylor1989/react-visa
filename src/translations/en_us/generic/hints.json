{
  "h1": "YOU CAN CHANGE YOUR REGION LATER IN THE SETTINGS PANEL IN THE BOTTOM RIGHT OF THE SCREEN.",
  "h2": "TO GET STARTED, SELECT A PRODUCT YOU'D LIKE TO LEARN MORE ABOUT.",
  "h3": "WANT TO EARN BONUS POINTS? RETRY ANY PRODUCT CHALLENGE AND TRY TO GET A PERFECT SCORE.",
  "h4": "YOU CAN EARN MORE POINTS BY COMPLETING MORE PRODUCT CHALLENGES.",
  "h5": "SELECT MORE ON THE DASHBOARD TO SEE THE FULL ACHIEVEMENT, RANK, PRODUCT LIST.",
  "h6": "EARN THE DAILY STREAK BONUS BY COMPLETING 4 CHALLENGES IN A ROW.",
  "h7": "SELECT CLIENT PROPOSALS TO PUT YOUR SOLUTIONS KNOWLEDGE TO THE TEST.",
  "h8": "KEEP GOING! YOU'VE NEARLY EARNED YOUR DAILY STREAK BONUS!",
  "h9": "YOU CAN GET A PERFECT BONUS BY SCORING 100% IN A CHALLENGE.",
  "h10": "SELECT THE BACK BUTTON TO RETURN TO THE PRODUCT NETWORK.",
  "h11": "YOU’LL NEED AN EYE FOR DETAIL AND PLENTY OF PRODUCT KNOWLEDGE TO WIN THESE CLIENT PROPOSALS.",
  "h12": "YOU’LL NEED TO REVIEW ALL THE RESOURCES TO DETERMINE THIS CLIENT’S NEEDS.",
  "h13": "FIND THE PERFECT SOLUTION FOR THIS CLIENT BY MATCHING THEIR NEEDS TO OUR PRODUCTS."
}
