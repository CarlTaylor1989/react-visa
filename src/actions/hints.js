import * as types from './hintsTypes';

export function setCurrentHintGroup(group) {
  return {
    type: types.SET_CURRENT_HINT_GROUP,
    group
  };
}

export function setHintGroupIndex(hints) {
  return {
    type: types.SET_HINT_GROUP_INDEX,
    hints
  };
}

export function setHintsPaused(paused) {
  return {
    type: types.SET_HINTS_PAUSED,
    paused
  };
}
