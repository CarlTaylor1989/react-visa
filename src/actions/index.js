import { push } from 'connected-react-router';
import * as AchievementActions from './achievements';
import * as AudioActions from './audio';
import * as ClientActions from './client';
import * as DiagnosticActions from './diagnostic';
import * as GenericActions from './generic';
import * as HintsActions from './hints';
import * as MapActions from './map';
import * as PromptActions from './prompts';
import * as ScoreActions from './score';
import * as ScormActions from './scorm';
import * as SettingsActions from './settings';
import * as StreakActions from './streak';
import * as TutorialActions from './tutorial';

const ActionCreators = Object.assign(
  {},
  AchievementActions,
  AudioActions,
  ClientActions,
  DiagnosticActions,
  GenericActions,
  HintsActions,
  MapActions,
  PromptActions,
  ScoreActions,
  ScormActions,
  SettingsActions,
  StreakActions,
  TutorialActions,
  { push }
);

export default ActionCreators;
