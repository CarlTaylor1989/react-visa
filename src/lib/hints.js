import hints, * as lib from '../config/hints';

/**
 * Filters groups by group into a new array.
 * @param {array} group Array of hints
 * @return {array}
 */
export const getHints = group => hints.filter(hint => hint.group === group);

/**
 * Gets the hint id for a hint based on it's group and index within that group.
 * @param {string} group Group name
 * @param {number} index Current index
 * @return {string | null}
 */
export const getHintId = (group, index) => {
  const hint = getHints(group)[index];
  return hint ? hint.id : null;
};

/**
 * Calculates whether a network hint is accessible.
 * @param {string} hintId Hint identifier
 * @param {object} state Game state
 * @return {boolean}
 */
export const networkHintAccessible = (hintId, state) => {
  let accessible = false;
  switch (hintId) {
    case 'h2':
      accessible = state.completedChallenges.length === 0;
      break;
    case 'h3':
    case 'h4':
    case 'h5':
    case 'h6':
      accessible = state.completedChallenges.length > 0;
      break;
    case 'h7':
      accessible = state.completedProducts.length > 0;
      break;
    case 'h8':
      accessible = state.bonusStreak === 3;
      break;
    default:
      accessible = true;
      break;
  }

  return accessible;
};

/**
 * Calculates whether a product hint is accessible.
 * @param {string} hintId Hint identifier
 * @param {object} state Game state
 * @return {boolean}
 */
export const productHintAccessible = (hintId, state) => {
  let accessible = false;
  switch (hintId) {
    case 'h9':
      accessible = state.completedChallenges.length > 0;
      break;
    case 'h10':
      accessible = state.productPageInactive;
      break;
    default:
      accessible = true;
      break;
  }
  return accessible;
};

export const clientProposalHintAccesible = (hintId, state) => {
  let accessible = false;
  const {
    clients,
    clientShowing,
    displayReqFeedback
  } = state;
  const {
    reqCorrect
  } = clients.find(({ id }) => id === clientShowing);
  switch (hintId) {
    case 'h12':
      accessible = !reqCorrect || (reqCorrect && displayReqFeedback);
      break;
    case 'h13':
      accessible = (reqCorrect && !displayReqFeedback);
      break;
    default:
      accessible = true;
      break;
  }
  return accessible;
};

/**
 * Calculates whether a hint is accessible based on game state. Custom logic for many of the hints.
 * @param {string} group Group name
 * @param {string} hintId Hint identifier
 * @param {object} state Game state
 * @return {boolean}
 */
export const hintAccessible = (group, hintId, state) => {
  let accessible = false;
  if (typeof hintId === 'string') {
    switch (group) {
      case lib.HINT_NETWORK:
        accessible = networkHintAccessible(hintId, state);
        break;
      case lib.HINT_PRODUCT:
        accessible = productHintAccessible(hintId, state);
        break;
      case lib.HINT_CLIENT_PROPOSAL:
        accessible = clientProposalHintAccesible(hintId, state);
        break;
      case lib.HINT_DIAGNOSTIC:
      case lib.HINT_CLIENT_SELECTION:
      default:
        accessible = true;
        break;
    }
  }
  return accessible;
};

/**
 * Recursive function that finds the next valid array index from the
 * array of hints filtered by group, this function takes note of the starting index
 * and returns it if it is reached again as to not end up in a infinite loop of recursive
 * function calls if no hints are currently available for a hint group.
 *
 * @param {string} group Group name
 * @param {number} index Current index
 * @param {object} state Game state
 * @param {number} startingIndex Starting index
 * @return {number}
 */
const getNextHintIndexRecursive = (group, index, state, startingIndex) => {
  const size = getHints(group).length - 1;
  const nextIndex = (index + 1 > size) ? 0 : index + 1;
  const hintId = getHintId(group, nextIndex);
  return (hintAccessible(group, hintId, state) || nextIndex === startingIndex)
    ? nextIndex
    : getNextHintIndexRecursive(group, nextIndex, state, startingIndex);
};

/**
 * Function to find the next valid array index from the
 * array of hints filtered by group.
 * @param {string} group Group name
 * @param {number} index Current index
 * @param {object} state Game state
 * @return {number}
 */
export const getNextHintIndex = (group, index, state) => getNextHintIndexRecursive(
  group,
  index,
  state,
  index
);

/**
 * Gets the current hint index for the current hint group from
 * game state.
 * @param {array} hintsState Array of hints from game state
 * @param {string} group Group name from game state
 * @return {number}
 */
export const getHintIndexFromState = (hintsState, group) => {
  const currentGroup = hintsState.find(hint => hint.group === group);
  return (currentGroup) ? currentGroup.index : null;
};
