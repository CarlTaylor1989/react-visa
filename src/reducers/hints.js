import { createReducer } from 'redux-create-reducer';
import * as types from '../actions/hintsTypes';
import * as lib from '../config/hints';

export const initialState = {
  currentGroup: '',
  paused: false,
  hints: [
    { group: lib.HINT_DIAGNOSTIC, index: 0 },
    { group: lib.HINT_NETWORK, index: 0 },
    { group: lib.HINT_PRODUCT, index: 0 },
    { group: lib.HINT_CLIENT_SELECTION, index: 0 },
    { group: lib.HINT_CLIENT_PROPOSAL, index: 0 }
  ]
};

export const hintsData = createReducer(Object.assign(initialState), {
  [types.SET_CURRENT_HINT_GROUP](state, action) {
    return {
      ...state,
      currentGroup: action.group
    };
  },
  [types.SET_HINT_GROUP_INDEX](state, action) {
    return {
      ...state,
      hints: state.hints.map(hint => (
        hint.group === action.hints.group ? {
          ...hint,
          index: action.hints.index
        } : hint))
    };
  },
  [types.SET_HINTS_PAUSED](state, action) {
    return {
      ...state,
      paused: action.paused
    };
  }
});

export default { hintsData };
