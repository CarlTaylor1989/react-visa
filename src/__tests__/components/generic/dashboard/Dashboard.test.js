/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import {
  Dashboard,
  mapDispatchToProps
} from '../../../../components/generic/dashboard/Dashboard';
import * as hintActions from '../../../../actions/hints';
import * as hintTypes from '../../../../actions/hintsTypes';

let mockProps;
beforeEach(() => {
  jest.resetAllMocks();
  mockProps = {
    closeDashboard: jest.fn(),
    displayDashboard: true,
    setHintsPaused: jest.fn()
  };
});

test('renders a closed Dashboard popup', () => {
  const wrapper = shallow(<Dashboard {...mockProps} displayDashboard={false} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('renders an open Dashboard popup', () => {
  const wrapper = shallow(<Dashboard {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

describe('method getContent()', () => {
  test('AchievementsView component', () => {
    const wrapper = shallow(<Dashboard {...mockProps} />);
    wrapper.instance().setState({ page: 'achievements' });
    expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
  });

  test('RankView component', () => {
    const wrapper = shallow(<Dashboard {...mockProps} />);
    const page = 'rank';
    wrapper.instance().setState({ page });
    expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
  });

  test('ProductsCompleteView component', () => {
    const wrapper = shallow(<Dashboard {...mockProps} />);
    wrapper.instance().setState({ page: 'productsCompleted' });
    expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
  });
});

test('method setView()', () => {
  const wrapper = shallow(<Dashboard {...mockProps} />);
  const page = 'rank';
  wrapper.instance().setView(page);
  expect(wrapper.instance().state.page).toEqual(page);
});

test('method setView() is called from achievements button click', () => {
  const wrapper = shallow(<Dashboard {...mockProps} />);
  wrapper.instance().setView = jest.fn();
  wrapper.find('.achievementsBtn').first().simulate('click');
  expect(wrapper.instance().setView).toHaveBeenCalledWith('achievements');
});

test('method setView() is called from rank button click', () => {
  const wrapper = shallow(<Dashboard {...mockProps} />);
  wrapper.instance().setView = jest.fn();
  wrapper.find('.rankBtn').first().simulate('click');
  expect(wrapper.instance().setView).toHaveBeenCalledWith('rank');
});

test('method setView() is called from products completed button click', () => {
  const wrapper = shallow(<Dashboard {...mockProps} />);
  wrapper.instance().setView = jest.fn();
  wrapper.find('.productsCompletedBtn').first().simulate('click');
  expect(wrapper.instance().setView).toHaveBeenCalledWith('productsCompleted');
});

describe('should mapDispatchToProps', () => {
  test('SET_HINTS_PAUSED', () => {
    const dispatch = jest.fn();
    jest.spyOn(hintActions, 'setHintsPaused').mockReturnValue('setHintsPaused()');
    const props = mapDispatchToProps(dispatch);
    props.setHintsPaused();
    expect(dispatch).toHaveBeenCalledWith({
      type: hintTypes.SET_HINTS_PAUSED
    });
  });
});
