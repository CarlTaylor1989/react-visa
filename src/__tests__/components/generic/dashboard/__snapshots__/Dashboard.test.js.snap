// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`method getContent() AchievementsView component 1`] = `
<Modal
  animationDuration={500}
  blockScroll={true}
  center={true}
  classNames={
    Object {
      "modal": "dashboard",
      "overlay": "dashboardOverlay",
    }
  }
  classes={
    Object {
      "closeButton": "styles_closeButton__20ID4",
      "closeIcon": "styles_closeIcon__1QwbI",
      "modal": "styles_modal__gNwvD",
      "modalCenter": "styles_modalCenter__L9F2w",
      "overlay": "styles_overlay__CLSq-",
      "transitionEnter": "styles_transitionEnter__3j_-a",
      "transitionEnterActive": "styles_transitionEnterActive___eQs7",
      "transitionExit": "styles_transitionExit__1KmEf",
      "transitionExitActive": "styles_transitionExitActive__1nQXw",
    }
  }
  closeIconSize={28}
  closeIconSvgPath={
    <path
      d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
    />
  }
  closeOnEsc={false}
  closeOnOverlayClick={false}
  container={null}
  focusTrapOptions={Object {}}
  focusTrapped={true}
  onClose={[MockFunction]}
  open={true}
  role="dialog"
  showCloseIcon={false}
  styles={Object {}}
>
  <button
    className="exitPopup"
    onClick={[MockFunction]}
    type="button"
  >
    <Translate
      tag="span"
      value="generic.popupCloseBtn"
    />
  </button>
  <div
    className="popupWrapper"
  >
    <div
      className="popupTitle"
    >
      <Translate
        tag="span"
        value="generic.dashboard.title"
      />
    </div>
    <div
      className="main"
    >
      <div
        className="topBar"
      >
        <div
          className="navigation"
        >
          <Connect(Tooltip)
            fieldName="generic.tooltips.dashboard.achievements.button"
          >
            <button
              className="achievementsBtn selected"
              onClick={[Function]}
              type="button"
            >
              <Translate
                tag="span"
                value="generic.dashboard.achievementsBtn"
              />
            </button>
          </Connect(Tooltip)>
          <Connect(Tooltip)
            fieldName="generic.tooltips.dashboard.rank.button"
          >
            <button
              className="rankBtn "
              onClick={[Function]}
              type="button"
            >
              <Translate
                tag="span"
                value="generic.dashboard.rankBtn"
              />
            </button>
          </Connect(Tooltip)>
          <Connect(Tooltip)
            fieldName="generic.tooltips.dashboard.products.button"
          >
            <button
              className="productsCompletedBtn "
              onClick={[Function]}
              type="button"
            >
              <Translate
                tag="span"
                value="generic.dashboard.productsCompletedBtn"
              />
            </button>
          </Connect(Tooltip)>
        </div>
        <div
          className="pointsContainer"
        >
          <Connect(Points) />
        </div>
      </div>
      <div
        className="content"
      >
        <Connect(AchievementsView) />
      </div>
    </div>
  </div>
</Modal>
`;

exports[`method getContent() ProductsCompleteView component 1`] = `
<Modal
  animationDuration={500}
  blockScroll={true}
  center={true}
  classNames={
    Object {
      "modal": "dashboard",
      "overlay": "dashboardOverlay",
    }
  }
  classes={
    Object {
      "closeButton": "styles_closeButton__20ID4",
      "closeIcon": "styles_closeIcon__1QwbI",
      "modal": "styles_modal__gNwvD",
      "modalCenter": "styles_modalCenter__L9F2w",
      "overlay": "styles_overlay__CLSq-",
      "transitionEnter": "styles_transitionEnter__3j_-a",
      "transitionEnterActive": "styles_transitionEnterActive___eQs7",
      "transitionExit": "styles_transitionExit__1KmEf",
      "transitionExitActive": "styles_transitionExitActive__1nQXw",
    }
  }
  closeIconSize={28}
  closeIconSvgPath={
    <path
      d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
    />
  }
  closeOnEsc={false}
  closeOnOverlayClick={false}
  container={null}
  focusTrapOptions={Object {}}
  focusTrapped={true}
  onClose={[MockFunction]}
  open={true}
  role="dialog"
  showCloseIcon={false}
  styles={Object {}}
>
  <button
    className="exitPopup"
    onClick={[MockFunction]}
    type="button"
  >
    <Translate
      tag="span"
      value="generic.popupCloseBtn"
    />
  </button>
  <div
    className="popupWrapper"
  >
    <div
      className="popupTitle"
    >
      <Translate
        tag="span"
        value="generic.dashboard.title"
      />
    </div>
    <div
      className="main"
    >
      <div
        className="topBar"
      >
        <div
          className="navigation"
        >
          <Connect(Tooltip)
            fieldName="generic.tooltips.dashboard.achievements.button"
          >
            <button
              className="achievementsBtn "
              onClick={[Function]}
              type="button"
            >
              <Translate
                tag="span"
                value="generic.dashboard.achievementsBtn"
              />
            </button>
          </Connect(Tooltip)>
          <Connect(Tooltip)
            fieldName="generic.tooltips.dashboard.rank.button"
          >
            <button
              className="rankBtn "
              onClick={[Function]}
              type="button"
            >
              <Translate
                tag="span"
                value="generic.dashboard.rankBtn"
              />
            </button>
          </Connect(Tooltip)>
          <Connect(Tooltip)
            fieldName="generic.tooltips.dashboard.products.button"
          >
            <button
              className="productsCompletedBtn selected"
              onClick={[Function]}
              type="button"
            >
              <Translate
                tag="span"
                value="generic.dashboard.productsCompletedBtn"
              />
            </button>
          </Connect(Tooltip)>
        </div>
        <div
          className="pointsContainer"
        >
          <Connect(Points) />
        </div>
      </div>
      <div
        className="content"
      >
        <ProductsCompleteView
          closePopup={[MockFunction]}
        />
      </div>
    </div>
  </div>
</Modal>
`;

exports[`method getContent() RankView component 1`] = `
<Modal
  animationDuration={500}
  blockScroll={true}
  center={true}
  classNames={
    Object {
      "modal": "dashboard",
      "overlay": "dashboardOverlay",
    }
  }
  classes={
    Object {
      "closeButton": "styles_closeButton__20ID4",
      "closeIcon": "styles_closeIcon__1QwbI",
      "modal": "styles_modal__gNwvD",
      "modalCenter": "styles_modalCenter__L9F2w",
      "overlay": "styles_overlay__CLSq-",
      "transitionEnter": "styles_transitionEnter__3j_-a",
      "transitionEnterActive": "styles_transitionEnterActive___eQs7",
      "transitionExit": "styles_transitionExit__1KmEf",
      "transitionExitActive": "styles_transitionExitActive__1nQXw",
    }
  }
  closeIconSize={28}
  closeIconSvgPath={
    <path
      d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
    />
  }
  closeOnEsc={false}
  closeOnOverlayClick={false}
  container={null}
  focusTrapOptions={Object {}}
  focusTrapped={true}
  onClose={[MockFunction]}
  open={true}
  role="dialog"
  showCloseIcon={false}
  styles={Object {}}
>
  <button
    className="exitPopup"
    onClick={[MockFunction]}
    type="button"
  >
    <Translate
      tag="span"
      value="generic.popupCloseBtn"
    />
  </button>
  <div
    className="popupWrapper"
  >
    <div
      className="popupTitle"
    >
      <Translate
        tag="span"
        value="generic.dashboard.title"
      />
    </div>
    <div
      className="main"
    >
      <div
        className="topBar"
      >
        <div
          className="navigation"
        >
          <Connect(Tooltip)
            fieldName="generic.tooltips.dashboard.achievements.button"
          >
            <button
              className="achievementsBtn "
              onClick={[Function]}
              type="button"
            >
              <Translate
                tag="span"
                value="generic.dashboard.achievementsBtn"
              />
            </button>
          </Connect(Tooltip)>
          <Connect(Tooltip)
            fieldName="generic.tooltips.dashboard.rank.button"
          >
            <button
              className="rankBtn selected"
              onClick={[Function]}
              type="button"
            >
              <Translate
                tag="span"
                value="generic.dashboard.rankBtn"
              />
            </button>
          </Connect(Tooltip)>
          <Connect(Tooltip)
            fieldName="generic.tooltips.dashboard.products.button"
          >
            <button
              className="productsCompletedBtn "
              onClick={[Function]}
              type="button"
            >
              <Translate
                tag="span"
                value="generic.dashboard.productsCompletedBtn"
              />
            </button>
          </Connect(Tooltip)>
        </div>
        <div
          className="pointsContainer"
        >
          <Connect(Points) />
        </div>
      </div>
      <div
        className="content"
      >
        <Connect(RankView) />
      </div>
    </div>
  </div>
</Modal>
`;

exports[`renders a closed Dashboard popup 1`] = `
<Modal
  animationDuration={500}
  blockScroll={true}
  center={true}
  classNames={
    Object {
      "modal": "dashboard",
      "overlay": "dashboardOverlay",
    }
  }
  classes={
    Object {
      "closeButton": "styles_closeButton__20ID4",
      "closeIcon": "styles_closeIcon__1QwbI",
      "modal": "styles_modal__gNwvD",
      "modalCenter": "styles_modalCenter__L9F2w",
      "overlay": "styles_overlay__CLSq-",
      "transitionEnter": "styles_transitionEnter__3j_-a",
      "transitionEnterActive": "styles_transitionEnterActive___eQs7",
      "transitionExit": "styles_transitionExit__1KmEf",
      "transitionExitActive": "styles_transitionExitActive__1nQXw",
    }
  }
  closeIconSize={28}
  closeIconSvgPath={
    <path
      d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
    />
  }
  closeOnEsc={false}
  closeOnOverlayClick={false}
  container={null}
  focusTrapOptions={Object {}}
  focusTrapped={true}
  onClose={[MockFunction]}
  open={false}
  role="dialog"
  showCloseIcon={false}
  styles={Object {}}
>
  <button
    className="exitPopup"
    onClick={[MockFunction]}
    type="button"
  >
    <Translate
      tag="span"
      value="generic.popupCloseBtn"
    />
  </button>
  <div
    className="popupWrapper"
  >
    <div
      className="popupTitle"
    >
      <Translate
        tag="span"
        value="generic.dashboard.title"
      />
    </div>
    <div
      className="main"
    >
      <div
        className="topBar"
      >
        <div
          className="navigation"
        >
          <Connect(Tooltip)
            fieldName="generic.tooltips.dashboard.achievements.button"
          >
            <button
              className="achievementsBtn selected"
              onClick={[Function]}
              type="button"
            >
              <Translate
                tag="span"
                value="generic.dashboard.achievementsBtn"
              />
            </button>
          </Connect(Tooltip)>
          <Connect(Tooltip)
            fieldName="generic.tooltips.dashboard.rank.button"
          >
            <button
              className="rankBtn "
              onClick={[Function]}
              type="button"
            >
              <Translate
                tag="span"
                value="generic.dashboard.rankBtn"
              />
            </button>
          </Connect(Tooltip)>
          <Connect(Tooltip)
            fieldName="generic.tooltips.dashboard.products.button"
          >
            <button
              className="productsCompletedBtn "
              onClick={[Function]}
              type="button"
            >
              <Translate
                tag="span"
                value="generic.dashboard.productsCompletedBtn"
              />
            </button>
          </Connect(Tooltip)>
        </div>
        <div
          className="pointsContainer"
        >
          <Connect(Points) />
        </div>
      </div>
      <div
        className="content"
      >
        <Connect(AchievementsView) />
      </div>
    </div>
  </div>
</Modal>
`;

exports[`renders an open Dashboard popup 1`] = `
<Modal
  animationDuration={500}
  blockScroll={true}
  center={true}
  classNames={
    Object {
      "modal": "dashboard",
      "overlay": "dashboardOverlay",
    }
  }
  classes={
    Object {
      "closeButton": "styles_closeButton__20ID4",
      "closeIcon": "styles_closeIcon__1QwbI",
      "modal": "styles_modal__gNwvD",
      "modalCenter": "styles_modalCenter__L9F2w",
      "overlay": "styles_overlay__CLSq-",
      "transitionEnter": "styles_transitionEnter__3j_-a",
      "transitionEnterActive": "styles_transitionEnterActive___eQs7",
      "transitionExit": "styles_transitionExit__1KmEf",
      "transitionExitActive": "styles_transitionExitActive__1nQXw",
    }
  }
  closeIconSize={28}
  closeIconSvgPath={
    <path
      d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
    />
  }
  closeOnEsc={false}
  closeOnOverlayClick={false}
  container={null}
  focusTrapOptions={Object {}}
  focusTrapped={true}
  onClose={[MockFunction]}
  open={true}
  role="dialog"
  showCloseIcon={false}
  styles={Object {}}
>
  <button
    className="exitPopup"
    onClick={[MockFunction]}
    type="button"
  >
    <Translate
      tag="span"
      value="generic.popupCloseBtn"
    />
  </button>
  <div
    className="popupWrapper"
  >
    <div
      className="popupTitle"
    >
      <Translate
        tag="span"
        value="generic.dashboard.title"
      />
    </div>
    <div
      className="main"
    >
      <div
        className="topBar"
      >
        <div
          className="navigation"
        >
          <Connect(Tooltip)
            fieldName="generic.tooltips.dashboard.achievements.button"
          >
            <button
              className="achievementsBtn selected"
              onClick={[Function]}
              type="button"
            >
              <Translate
                tag="span"
                value="generic.dashboard.achievementsBtn"
              />
            </button>
          </Connect(Tooltip)>
          <Connect(Tooltip)
            fieldName="generic.tooltips.dashboard.rank.button"
          >
            <button
              className="rankBtn "
              onClick={[Function]}
              type="button"
            >
              <Translate
                tag="span"
                value="generic.dashboard.rankBtn"
              />
            </button>
          </Connect(Tooltip)>
          <Connect(Tooltip)
            fieldName="generic.tooltips.dashboard.products.button"
          >
            <button
              className="productsCompletedBtn "
              onClick={[Function]}
              type="button"
            >
              <Translate
                tag="span"
                value="generic.dashboard.productsCompletedBtn"
              />
            </button>
          </Connect(Tooltip)>
        </div>
        <div
          className="pointsContainer"
        >
          <Connect(Points) />
        </div>
      </div>
      <div
        className="content"
      >
        <Connect(AchievementsView) />
      </div>
    </div>
  </div>
</Modal>
`;
